### PR TITLE
utfcpp: new package at 3.1

### PR DESCRIPTION
**Pull request by eudaldgr**
## Description of package

[UTF-8 with C++ in a Portable Way](https://github.com/nemtrif/utfcpp)

## Installed manifest of package

```
/var/db/kiss/installed/utfcpp/version
/var/db/kiss/installed/utfcpp/sources
/var/db/kiss/installed/utfcpp/manifest
/var/db/kiss/installed/utfcpp/depends
/var/db/kiss/installed/utfcpp/checksums
/var/db/kiss/installed/utfcpp/build
/var/db/kiss/installed/utfcpp/
/var/db/kiss/installed/
/var/db/kiss/
/var/db/
/var/
/usr/lib/cmake/utf8cpp/utf8cppConfig.cmake
/usr/lib/cmake/utf8cpp/
/usr/lib/cmake/
/usr/lib/
/usr/include/utf8cpp/utf8/unchecked.h
/usr/include/utf8cpp/utf8/cpp11.h
/usr/include/utf8cpp/utf8/core.h
/usr/include/utf8cpp/utf8/checked.h
/usr/include/utf8cpp/utf8/
/usr/include/utf8cpp/utf8.h
/usr/include/utf8cpp/
/usr/include/
/usr/
```

## New package

- [x] Latest upstream version.
- [x] I agree to maintain this package.
- [x] I agree to notify @dylanaraps if I can no longer maintain this package.

### CODE REVIEW

File: community/utfcpp/build [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh -e
+
+mkdir -p build
+cd build
+
+cmake .. -DUTF8_TESTS=Off -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr
+
+make
+make DESTDIR="$1" install
```
Comment (dylanaraps) [inline community/utfcpp/build]: Please read https://k1ss.org/style

**Comment by dylanaraps** [comment]
Also. What is this library for? Will you submit a package which will depend on it?

**Comment by eudaldgr** [comment]
Later i will submit a package which depend on it.

File: community/utfcpp/build (hunk 1 above)
Comment (eudaldgr) [reply community/utfcpp/build]: Have I understood correctly?

Comment (dylanaraps) [reply community/utfcpp/build]: See 0800

Comment (eudaldgr) [reply community/utfcpp/build]: Sorry, i'm learning.

File: community/utfcpp/build [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh -e
+
+export DESTDIR="$1"
+
+cmake -B build \
+    -DUTF8_TESTS=off \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr
```
Comment (dylanaraps) [inline community/utfcpp/build]: Reverse this list.

```
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DCMAKE_BUILD_TYPE=Release \
    -DUTF8_TESTS=off
```

**Comment by dylanaraps** [comment]
```
/usr/lib64/cmake/utf8cpp/utf8cppConfig.cmake
/usr/lib64/cmake/utf8cpp/
/usr/lib64/cmake/
/usr/lib64/
```

These should be in `/usr/lib` and not `/usr/lib64`.

**Comment by eudaldgr** [comment]
I have in `lib64` yet

**Comment by eudaldgr** [comment]
> ```
> /usr/lib64/cmake/utf8cpp/utf8cppConfig.cmake
> /usr/lib64/cmake/utf8cpp/
> /usr/lib64/cmake/
> /usr/lib64/
> ```
> 
> These should be in `/usr/lib` and not `/usr/lib64`.

I tried `-DLIBDIR=/usr/lib \` but this library don't use this variable.

**Comment by dylanaraps** [comment]
This is something you will have to fix, otherwise I won't be able to merge this.

**Comment by konimex** [comment]
Try `-DCMAKE_INSTALL_LIBDIR`.

**Comment by dylanaraps** [comment]
Thanks, will merge once your other PRs are open.

**Comment by dylanaraps** [comment]
Ledger fails to link with our boost. Have you not built the package yourself yet?

```
/usr/lib/gcc/x86_64-pc-linux-musl/10.0.1/../../../../x86_64-pc-linux-musl/bin/ld: /usr/lib64/libboost_filesystem.a(directory.o): relocation R_X86_64_PC32 against symbol `_ZN5boost6system6detail10cat_holderIvE24system_category_instanceE' can not be used when making a shared object; recompile with -fPIC
/usr/lib/gcc/x86_64-pc-linux-musl/10.0.1/../../../../x86_64-pc-linux-musl/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
ninja: subcommand failed
```

**Comment by eudaldgr** [comment]
I'm trying right now, with the dependencies you added.

**Pull request closed without merging**